util/testutil/error.go

```diff
@@ -13,15 +13,11 @@
 
 package testutil
 
-// ErrorEqual compares Go errors for equality.
-func ErrorEqual(left, right error) bool {
-	if left == right {
-		return true
+// ErrorsEqual fails when Go errors are not equal.
+func ErrorsEqual(tb TB, exp, act error) {
+	if exp != nil && act != nil {
```

Comment:
We should fail even when one of them is `nil` and the other is not. You could follow the old implementation and just replace the `return false` cases with `tb.Fatalf`.

Reply:
Yes, good point. Fixed.